### PR TITLE
Add `newWindow` option

### PR DESCRIPTION
**Pull request by jebrii**
Very light feature add here. Added an option for opening an application in a new window. This is useful for when the application is already open.

For example, if attempting to open an incognito window in Google Chrome (see #41 ), the proper command in a Mac OS terminal would be:

`open -n -a "Google Chrome" "$url" --args --incognito`

This should solve #41 by allowing users to use this tool to execute that command.

--------------------

EDIT: upon further testing, I've found that the url param in the above command (`"$url"`) must be placed last:
`open -n -a "Google Chrome" --args --incognito "$url"`

I'm adding that change to this PR.

### CODE REVIEW

**Comment by MortenGregersen** [comment]
It looks like indention is broken on the lines added.

**Comment by jebrii** [comment]
> It looks like indention is broken on the lines added.

Thanks for bringing this to my attention!

File: index.d.ts [hunk 1]
```diff
@@ -23,6 +23,15 @@ declare namespace open {
 		*/
 		readonly background?: boolean;
 
+		/**
+		__macOS only__
+
+		Open the specified url in a new window. This is useful when the app is expected to already be open.
```
Comment (sindresorhus) [inline index.d.ts:29]: `open` doesn't only open URLs. Also `url` => `URL`.

File: index.js [hunk 2]
```diff
@@ -65,9 +66,17 @@ module.exports = async (target, options) => {
 			cliArguments.push('--background');
 		}
 
+		if (options.newWindow) {
+			cliArguments.push('-n');
```
Comment (sindresorhus) [inline index.js:70]: Use the long-flag.

File: index.js [hunk 3]
```diff
@@ -65,9 +66,17 @@ module.exports = async (target, options) => {
...
+		}
+
 		if (app) {
 			cliArguments.push('-a', app);
 		}
+
+		if (appArguments.length > 0) {
+			cliArguments.push('--args', ...appArguments);
+		}
```
Comment (sindresorhus) [inline index.js:79]: Are you sure moving it here is correct? I think the `target` (https://github.com/sindresorhus/open/pull/206/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R155) is order dependent.

File: package.json [hunk 4]
```diff
@@ -48,6 +48,7 @@
 		"file"
 	],
 	"dependencies": {
+		"child_process": "^1.0.2",
```
Comment (sindresorhus) [inline package.json:51]: This is incorrect.

File: index.d.ts [hunk 5]
```diff
@@ -23,6 +23,15 @@ declare namespace open {
...
+
```
Comment (sindresorhus) [inline index.d.ts:30]: Mention your specific use-case of opening incognito mode.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**